compatibility with clang-format 20

Coeur opened the pull request:
Currently it's not easily possible to contribute to Transmission from a Mac.
On macOS, when using brew, only clang-format 20 is supported: we can't install clang-format 17 easily.
```
✗ brew search clang-format
==> Formulae
clang-format ✔                                              clang-format@11                                             cfn-format
```
In order to be able to make commits and contribute on Transmission, we need to pass the clang-format check.
Attempting `clang-format -i`.

Coeur commented:
I consider it a Transmission issue to enforce an obsolete tool.

Coeur commented:
Note: nowadays, only **clang-format 21.1** is supported on macOS.

mikedld commented:
@Coeur,

> I've been purposely delaying doing most code review until committing on macOS is fixed. It's a blocker for me to be unable to `git commit` properly:
> …
> Note: nowadays, only **clang-format 21.1** is supported on macOS.

_(quote from another unrelated PR)_

Tying the version of clang-format we use to whatever is available on a particular system via a particular package manager is an impossible task — simply because there's not just one system and one package manager, and they all never provide the same versions. And because different projects may require different versions, we sometimes even need to use multiple versions of clang-format on the same system depending on the project (or even within the same project, if contributing a fix to an old branch is required).

There're projects like https://github.com/muttleyxd/clang-tools-static-binaries which provide prebuilt binaries for earlier versions. You can even build clang-format yourself, if you want to.

All I'm saying is, if you really wanted to contribute, you'd find a way. Sorry if it sounds harsh, but this artificial requirement and demands linked to it don't look nice. Of course it'd be very nice if the version matched whatever you can easily install, but it's not a blocker for your work, and definitely not a blocked for core reviews where no code formatting is even necessary on your side.

ckerr commented:
Hmm one part of my last comment is not correct. Looks like there are some PRs that are approved but aren't showing up in that list and I'm not sure why. I'll take a look

Coeur commented:
@ckerr @tearfur 
Thank you for finally merging most of the changes of this PR via https://github.com/transmission/transmission/pull/7573.

All that is left now is the compatibility with clang-format 21.

Kindly review and merge the remaining changes of my pull request so that macOS contributions can resume, thanks.

Coeur commented:
So no one is willing to unblock the situation for macOS contributors.
It's 5-6 trivial lines to review.

ckerr reviewed (commented):
Can we add a note to remove these when we finally bump clang-format?

```diff
-         // clang-format off
+         // clang-format off: TODO: remove when we bump to clang-format >= 21
```

> Note: nowadays, only clang-format 21.1 is supported on macOS.

Frustratingly, GitHub's [Ubuntu 24.04 runners](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2404-Readme.md#ubuntu-2404) are still using clang-format <= 18.1.3

Coeur commented:
> Can we add a note to remove these when we finally bump clang-format?

Sure. I added a note for all the "clang-format off" of libtransmission in ef30b0ff44c7b3c59532e3257b66306daa27d2f7. And I removed the superfluous ones.
